pennylane/transforms/adjoint.py

```diff
@@ -102,9 +102,10 @@ def wrapper(*args, **kwargs):
             try:
                 op.adjoint()
             except NotImplementedError:
-                # Decompose the operation and adjoint the result.
+                # Expand the operation and adjoint the result.
                 # We do not do anything with the output since
-                # decomposition will automatically queue the new operations.
-                adjoint(op.decomposition)(wires=op.wires)
+                # calling adjoint on the expansion will automatically
+                # queue the new operations.
+                adjoint(op.expand)()

```

Comment:
Much cleaner, because we do not need to know a signature of `decomposition` here.